src/lib/cache/cacheManager.ts

```diff
@@ -0,0 +1,356 @@
+}
+
+export async function loadSinglePrefixCommandChannelDefaultVersionToCache(channelDefaultVersion: IPrefixCommandChannelDefaultVersion) {
+    const inMemoryCache = getInMemoryCache();
+    if (!inMemoryCache) return;
+
+    const { channelId, versionId } = channelDefaultVersion;
+    const version = await PrefixCommandVersion.findById(versionId);
+    if (version) {
+        Logger.debug(`Loading default version for channel ${channelId} to cache`);
+        await inMemoryCache.set(`${memoryCachePrefixChannelDefaultVersion}:${channelId}`, version.toObject());
+    }
+}
+
+export async function loadAllPrefixCommandChannelDefaultVersionsToCache() {
+    const conn = getConn();
+    const inMemoryCache = getInMemoryCache();
+    if (!conn || !inMemoryCache) return;
+
+    const PrefixCommandChannelDefaultVersions = await PrefixCommandChannelDefaultVersion.find();
+
+    for (const defaultVersion of PrefixCommandChannelDefaultVersions) {
+        // eslint-disable-next-line no-await-in-loop
+        await loadSinglePrefixCommandChannelDefaultVersionToCache(defaultVersion);
+    }
+}
+
+export async function refreshAllPrefixCommandChannelDefaultVersionsCache() {
+    const conn = getConn();
+    const inMemoryCache = getInMemoryCache();
+    if (!conn || !inMemoryCache) return;
+
+    // Step 1: Get all channel default versions from the database
+    const prefixCommandChannelDefaultVersions = await PrefixCommandChannelDefaultVersion.find();
+    // Step 2: Get all channel default versions from the cache
+    const cacheKeys = await inMemoryCache.store.keys();
+    // Step 3: Loop over cached channel default versions
+    for (const key of cacheKeys) {
+        if (key.startsWith(`${memoryCachePrefixChannelDefaultVersion}:`)) {
+            const channelId = key.split(':')[1];
+            // Step 3.a: Check if cached channel default version exists in the database list
+            let found = false;
+            for (const dbChannelDefaultVersion of prefixCommandChannelDefaultVersions) {
+                const { channelId: dbChannelId } = dbChannelDefaultVersion;
+                if (dbChannelId.toString().toLowerCase() === channelId.toLowerCase()) {
+                    found = true;
+                    break;
+                }
+            }
+            // Step 3.b: If not found, remove from cache
+            if (!found) {
+                Logger.debug(`Removing channel default version for channel ${channelId} from cache`);
+                // eslint-disable-next-line no-await-in-loop
+                await inMemoryCache.del(key);
+            }
+        }
+    }
+    // Step 4: Loop over database channel default versions and update cache
+    for (const dbChannelDefaultVersion of prefixCommandChannelDefaultVersions) {
+        // eslint-disable-next-line no-await-in-loop
+        await loadSinglePrefixCommandChannelDefaultVersionToCache(dbChannelDefaultVersion);
+    }
```

Comment:
```suggestion
    await Promise.all(
        prefixCommandChannelDefaultVersions
            .map(
                (dbChannelDefaultVersion) => loadSinglePrefixCommandChannelDefaultVersionToCache(dbChannelDefaultVersion),
            ),
    );
```
Even longer lmao. But I'm always a fan of descriptive names so I rather take that than guessing what a function may do 👍.